packages/vats/src/lib-board.js

```diff
@@ -1,22 +1,55 @@
 // @ts-check
+/**
+ * @file lib-board: share objects by way of plain-data ids
+ *
+ * @see prepareBoardClass()
+ */
 
-import { assert, details as X, q, Fail } from '@agoric/assert';
+import { assert, q, Fail } from '@agoric/assert';
 import { E, Far } from '@endo/far';
 import { makeMarshal } from '@endo/marshal';
-import { makeScalarMapStore } from '@agoric/store';
-import { crc6 } from './crc.js';
+import {
+  M,
+  makeScalarBigMapStore,
+  prepareExoClassKit,
+  provide,
+} from '@agoric/vat-data';
+import { makeEphemeraProvider } from '@agoric/zoe/src/contractSupport/durability.js';
 
-import './types.js';
+import { crc6 } from './crc.js';
 
 export const DEFAULT_CRC_DIGITS = 2;
 export const DEFAULT_PREFIX = 'board0';
 
+// #region Interface Guards
+const CapDataShape = { body: M.string(), slots: M.array() };
+const MarshalI = M.interface('Marshaller', {
+  serialize: M.call(M.any()).returns(CapDataShape),
+  unserialize: M.call(CapDataShape).returns(M.any()),
+});
```

Comment:
don't we have these elsewhere? They're the concern of Endo so it should export them, but until it does I think they belong in /internal.

Reply:
turns this is awkward until `@endo/patterns` is available to agoric-sdk